GAE/src/org/akvo/flow/events/EventLogger.java

```diff
@@ -0,0 +1,242 @@
+            messageMap.put(Key.APP_ID, appId);
+            messageMap.put(Key.URL, appId + ".appspot.com");
+
+            ObjectMapper m = new ObjectMapper();
+            OutputStreamWriter writer = new OutputStreamWriter(connection.getOutputStream());
+            m.writeValue(writer, messageMap);
+            writer.close();
+            if (connection.getResponseCode() != HttpURLConnection.HTTP_NO_CONTENT) {
+                logger.log(Level.SEVERE, "Unified log notification failed with status code: "
+                        + connection.getResponseCode());
+            }
+        } catch (MalformedURLException e) {
+            logger.log(Level.SEVERE,
+                    "Unified log notification failed with malformed URL exception", e);
+        } catch (IOException e) {
+            logger.log(Level.SEVERE, "Unified log notification failed with IO exception", e);
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, "Unified log notification failed with error", e);
+        }
+    }
+
+    /*
+     * Notify the log that a new event is ready to be downloaded.
+     */
+    private void notifyLog() {
+        Cache cache = initCache(60 * 60); // 1 hour
+        if (cache == null) {
+            // cache is not accessible, so we will notify anyway
+            logger.log(Level.WARNING,
+                    "cache not accessible, but still sending notification to unified log");
+            sendNotification();
+            return;
+        }
+
+        if (cache.containsKey(Action.UNIFIED_LOG_NOTIFIED)) {
+            // check if the time the last notification was send is less than one minute ago
+            Date cacheDate = (Date) cache.get(Action.UNIFIED_LOG_NOTIFIED);
+            Date nowDate = new Date();
+            Long deltaMils = nowDate.getTime() - cacheDate.getTime();
+            if (deltaMils < MIN_TIME_DIFF) {
+                // it is too soon, so don't send the notification
+                return;
+            }
+        }
+        // if we are here, either the key is not in the cache, or it is too old
+        // in both cases, we send the notification and add a fresh value to the cache
+        sendNotification();
+        cache.put(Action.UNIFIED_LOG_NOTIFIED, new Date());
+    }
+
+    private void storeEvent(Map<String, Object> event, Date timestamp) {
+        try {
+            ObjectMapper m = new ObjectMapper();
+            StringWriter w = new StringWriter();
+            m.writeValue(w, event);
+            BaseDAO<EventQueue> eventDao = new BaseDAO<EventQueue>(EventQueue.class);
+            EventQueue eventQ = new EventQueue();
+            eventQ.setCreatedDateTime(timestamp);
+            eventQ.setPayload(w.toString());
+            eventDao.save(eventQ);
+            notifyLog();
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, "could not store " + event.get("eventType")
+                    + " event. Error: " + e.toString());
```

Comment:
We always want the call that uses, the `Throwable` as second parameter, as we want the full stack trace